db/seeds.rb

```diff
@@ -1,8 +1,27 @@
-supplier = Supplier.find_or_create_by!(name: 'Bobs Cheese Shop')
-due_date = Date.today.next_month.change(day: 7)
+def definition_source(number)
+  <<~FDL
+    Framework cboard#{number} {
+      Name 'Cheese Board #{number}'
+      ManagementCharge 0.5% of 'Supplier Price'
+      Lots { '99' -> 'Fake' }
+      InvoiceFields {
+        InvoiceValue from 'Supplier Price'
+      }
+    }
+  FDL
+end
+
+def create_framework(number)
+  source = definition_source(number)
+  Framework.new_from_fdl(source).tap do |framework|
```

Comment:
Before this seed script could be run multiple times even if records already existed through `find_or_create_by`, during my test I got validation errors where short name had already been taken. I wonder if it would be possible to preserve this? Or to clean the slate before each run and delete all frameworks. I would really be more comfortable if this script had a guard against being run in production, if we chose the latter that would be especially important.

Reply:
Good point - I think that it's definitely reasonable to expect that I can call `db:seed` twice and that the second call will:

1. succeed
2. not create new records

I have expanded the tests to capture this behaviour; take a look at see what you think.

We could ask further questions about what kind of behaviour we expect from our seeds. For example, what if:

- we seed the database, then modify some of the data that it created, then run the seed command again?
- we seed the database, modify the seeds file, and then run the seed command again (i.e. should `seeds.rb` always be compatible with old seed data?)

I don't know whether answering these needs to be within the scope of this PR, though. As for deleting everything each time we run the seed command, I think it would be better if we could avoid doing that; it might be a slightly surprising behaviour to developers.

Reply:
This now looks good to me 👍 

In general I think it would be valuable to keep our default seed as conventional as possible, that would prompt us make a new `rake db:fresh_seed` to wipe out any edits and start again if that was needed.